src/matlab2tikz.m

```diff
@@ -2930,15 +2939,13 @@ case size(cData,2) % specific y-data is given
 
             % Calculate the width of each bar and the center point shift as in
             % makebars.m
-            if isempty(m2t.barShifts)
-                % Get the shifts of the bar centers.
-                % In case of numBars==1, this returns 0,
-                % In case of numBars==2, this returns [-1/4, 1/4],
-                % In case of numBars==3, this returns [-1/3, 0, 1/3],
-                % and so forth.
-                barWidth = groupWidth/numBars; % assumption
-                m2t.barShifts = ((1:numBars) - 0.5) * barWidth - groupWidth/2;
-            end
+            % Get the shifts of the bar centers.
+            % In case of numBars==1, this returns 0,
+            % In case of numBars==2, this returns [-1/4, 1/4],
+            % In case of numBars==3, this returns [-1/3, 0, 1/3],
+            % and so forth.
+            barWidth = groupWidth/numBars; % assumption
+            barShift = (m2t.barplotId - 0.5) * barWidth - groupWidth/2;
```

Comment:
bar(rand(2,3))

![untitled](https://cloud.githubusercontent.com/assets/3731173/5691853/fb7c5784-98d2-11e4-88fa-2015459fc6e9.png)

It is simpler with an example. Here, `numBars` is 3, i.e. each group has 3 bars, not to be confounded with a total of two bars per bar series.